ocw_import/api.py

```diff
@@ -149,33 +143,45 @@ def update_ocw2hugo_content(
                 filepath = obj["Key"]
             try:
                 text_id, content_data = get_content_data(filepath, s3_content, website)
-
-                content_file = WebsiteContent.objects.filter(
-                    website=website, text_id=text_id
-                ).first()
-
-                if content_file:
-                    if is_metadata_field:
-                        set_dict_field(
-                            content_file.metadata,
-                            update_field,
-                            get_dict_field(
-                                content_data.get("metadata", {}), update_field
-                            ),
-                        )
-                    elif update_field is not None:
-                        setattr(
-                            content_file,
-                            update_field,
-                            content_data.get(update_field, ""),
-                        )
-                    content_file.save()
-                elif create_new_content is True:
+                content = update_content_from_s3_data(website=website, text_id=text_id, content_data=content_data, update_field=update_field)
+                if not content and create_new_content is True:
                     convert_data_to_content(filepath, s3_content, website)
 
             except:  # pylint:disable=bare-except
                 log.exception("Error saving WebsiteContent for %s", s3_key)
 
+def update_content_from_s3_data(website, text_id, content_data, update_field):
+    is_metadata_field = False
+
+    if update_field and update_field.startswith("metadata."):
+        is_metadata_field = True
+        update_field = update_field.replace("metadata.", "", 1)
+
+    content_file = WebsiteContent.objects.filter(
+        website=website, text_id=text_id
+    ).first()
+    
+    if not content_file:
+        return None
+
+    if is_metadata_field:
+        set_dict_field(
+            content_file.metadata,
+            update_field,
+            get_dict_field(
+                content_data.get("metadata", {}), update_field
+            ),
+        )
+        if update_field == "parent_uid" and content_data["parent"]:
+            content_file.parent_id = content_data["parent"].id
```

Comment:
This is the real change. I moved this all to a separate function partly for sanity and partly to appease the linting gods.